iothub/device/devdoc/API Design.md

```diff
@@ -0,0 +1,275 @@
+## Plug and Play convention compatible APIs
+
+#### Common
+
+```csharp
+
+public abstract class PayloadConvention {
+    protected PayloadConvention();
+    public abstract ContentEncoder PayloadEncoder { get; }
+    public abstract Serializer PayloadSerializer { get; }
+    public virtual byte[] GetObjectBytes(object objectToSendWithConvention);
+}
+
+public abstract class ContentEncoder {
+    protected ContentEncoder();
+    public abstract Encoding ContentEncoding { get; }
+    public abstract byte[] EncodeStringToByteArray(string contentPayload);
+}
+
+public abstract class Serializer {
+    protected Serializer();
+    public abstract string ContentType { get; }
+    public abstract bool CheckWritablePropertyResponseType(object typeToCheck);
+    public abstract T ConvertFromObject<T>(object objectToConvert);
+    public abstract T DeserializeToType<T>(string stringToDeserialize);
+    public abstract string SerializeToString(object objectToSerialize);
+}
+
+public sealed class DefaultPayloadConvention : PayloadConvention {
+    public static readonly DefaultPayloadConvention Instance;
```

Comment:
Why named Instance?

Reply:
Instead of instantiating the instance it made more sense to create a default constructed instance that way if things change down we can just edit the property in this class and context is preserved